core/primitives/impl/scale_codec_impl.cpp

```diff
@@ -305,6 +327,39 @@ namespace kagome::primitives {
     return decodeVariant<Invalid, Valid, Unknown>(stream);
   }
 
+  outcome::result<Buffer> ScaleCodecImpl::encodeInherentData(
+      const InherentData &inherentData) const {
+    auto& data = inherentData.getDataCollection();
+
+    std::vector<primitives::InherentIdentifier> ids;
+    ids.reserve(data.size());
+    std::vector<std::vector<uint8_t>> vals;
```

Comment:
Can't we do it with a vector of const-refs-wrappers in order to avoid copying vectors?

Reply:
Now it's problematic because of the way encodeCollection works. It won't accept a vector of refs. I will leave a TODO to solve it after SCALE is refactored.